### PR TITLE
Minor refactors - cleaning models

### LINKED ISSUE #476: [coverage] Use `no cover` instead of `no-cover`
## 🚀 Refactoring
As far as I know from
- the docs: https://coverage.readthedocs.io/en/latest/excluding.html
- the number of search results on GitHub
  - `pragma: no cover` https://github.com/search?l=Python&q=pragma%3A+no+cover&type=code
  - `pragma: no-cover` https://github.com/search?l=Python&q=pragma%3A+no-cover&type=code

, we should use `pragma: no cover` instead of `pragma: no-cover`. I guess there's no difference between them, but most of the projects employ `pragma: no cover` as in the docs. (This should not be a high priority, but for consistency, we can start to use `no cover`.)

### LINKED ISSUE #519: Minor refactors of the codebase
## 🚀 Refactoring
This is a tracking issue for minor refactors (which makes sure not to change existing APIs):
- pragma (see also: #476)
  - Use `# pragma: no cover` instead of `# pragma: no-cover`.
  - Place `# pragma: no cover` at the beginning of clauses.
- imports
  - Add missing `__all__` to `__init__.py` files.
  - Use if/else instead of `try/except` for optional import statements
  - Raise `ModuleNotFoundError` with helpful messages **when instantiating classes or calling functions** in order not to fail to import classes/functions.

### Progress
- [x] `pl_bolts/callbacks/` #477
- [ ] `pl_bolts/datamodules/` #523
- [ ] `pl_bolts/datasets/` #523
- [ ] `pl_bolts/metrics/` #523
- [ ] `pl_bolts/models/` #524
- [x] `pl_bolts/models/self_supervised/` #479 #482
- [ ] `pl_bolts/optimizers/` #523
- [ ] `pl_bolts/transforms/` #523
- [ ] `pl_bolts/utils/` #523

### Additional context
After addressing this issue, #427 will follow up for major refactoring in models, and I'll add the importing tests in #421.

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Closes #519 and closes #476.

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Comment by akihironitta** [comment]
@Borda This PR should be ready to go. Please have a look :)

File: pl_bolts/models/regression/linear_regression.py [hunk 1]
```diff
@@ -110,16 +110,17 @@ def add_model_specific_args(parent_parser):
...
 
     pl.seed_everything(1234)
 
     # create dataset
-    try:
+    if _SKLEARN_AVAILABLE:
         from sklearn.datasets import load_boston
-    except ModuleNotFoundError as err:
-        raise ModuleNotFoundError(  # pragma: no-cover
+    else:  # pragma: no cover
```
Comment (Borda) [inline pl_bolts/models/regression/linear_regression.py:120]: there an advantage or using "no cover" compare to "no-cover" I was thinking that the joined form is more like a tag :]

**Review by Borda — APPROVED** [review]
